src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -818,20 +819,46 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     }
                 }
 
-                string method;
-                var begin = scan;
-                if (!begin.GetKnownMethod(out method))
+                end.Take();
+
+                var methodStart = scan;
+                var methodEnd = methodStart;
+                if (methodEnd.Seek(ref _vectorSpaces, ref end) == -1)
                 {
-                    if (scan.Seek(ref _vectorSpaces, ref end) == -1)
-                    {
-                        RejectRequest(RequestRejectionReason.MissingSpaceAfterMethod);
-                    }
+                    RejectRequest(RequestRejectionReason.InvalidRequestLine, GetInvalidRequestLineString(start, end));
```

Comment:
We make the effort using LoggerMessage.Define in KestrelTrace to not even call ILogger.Log if a given LogLevel isn't enabled.

I think it makes sense for this too. I'm not sure how common invalid request lines are, but a quick check using ILogger.IsEnabled(logLevel) doesn't seem too onerous.